client.go

```diff
@@ -0,0 +1,32 @@
+package dp_redis
+
+import (
+	"github.com/go-redis/redis"
+	"time"
+)
+
+// Redis - structure for the redis client
+type Redis struct {
+	client *redis.Client
+	ttl    time.Duration
+}
+
+// Options - config options for the redis client
+type Options struct {
+	Addr     string
+	Password string
+	DB       int
+	ttl      time.Duration
+}
+
+// NewClient - returns new redis client with provided config options
+func NewClient(o Options) *Redis {
```

Comment:
There should be some validation in here checking:

- The address is not empty
- Password is not empty
- ttl is a sensible value.

Personally I think t's better to return an error and fail to start up if its not right. Applying defaults is 'helpful' but ultimately masks the fact that something is not configured correctly.

Reply:
That makes perfect sense. I'll be sure to add some validation in.